src/ts-ast.ts

```diff
@@ -477,17 +477,30 @@ export class UnionType {
     // Functions too.
     const deduped = [];
     const names = new Set();
```

Comment:
BTW, sets are ordered in javascript, so you probably don't need both array and set representations here.

Also, better to be explicit about the element type of a set by initing it as `new Set<string>()`

Reply:
`deduped` is in array of AST nodes, `names` is an array of strings, can't re-use. Updated types to be more clear.

Reply:
+1